### PR TITLE
LibWeb+LibGfx: Support using a border-radius with <img> tags and background-images  

### PR DESCRIPTION
**Pull request by MacDue**
**Screenshot:**

image
**Left:** LibWeb, **Right:** Google Chrome

This PR adds support for using `border-radius` with background images and `` tags.
This is done via the `BorderRadiusCornerClipper` which is general enough to add this corner clipping to any element, though I think this PR covers the most common use-cases. 

The border radii can of course be elliptical, and are rendered antialiased :^)

(See commit messages for more details)

---

Test `old_computer.png` image is CC0.

---

> Would you look at that? Eight commits, interesting :thinking:

### CODE REVIEW

**Comment by MacDue** [comment]
CI Flakes in compiler cache :pensive:

File: Userland/Libraries/LibWeb/Painting/BorderRadiusCornerClipper.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#pragma once
+
+#include <LibGfx/AntiAliasingPainter.h>
+#include <LibWeb/Painting/BorderPainting.h>
+
+namespace Web::Painting {
+
+class BorderRadiusCornerClipper {
+public:
+    static ErrorOr<BorderRadiusCornerClipper> try_create(Gfx::IntRect const& border_rect, BorderRadiiData const& border_radii);
```
Comment (linusg) [inline Userland/Libraries/LibWeb/Painting/BorderRadiusCornerClipper.h:16]: `try_foo` is reserved for when there's already an infallible `foo` method that hasn't been removed yet

File: Userland/Libraries/LibWeb/Painting/BorderRadiusCornerClipper.h [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        CornerLocations bitmap_locations;
+        Gfx::IntSize corner_bitmap_size;
+    } m_data;
+
+    NonnullRefPtr<Gfx::Bitmap> m_corner_bitmap;
+    bool m_has_sampled { false };
+
+    BorderRadiusCornerClipper(CornerData corner_data, NonnullRefPtr<Gfx::Bitmap> corner_bitmap)
+        : m_data { corner_data }
+        , m_corner_bitmap { corner_bitmap }
```
Comment (linusg) [inline Userland/Libraries/LibWeb/Painting/BorderRadiusCornerClipper.h:46]: We don't use this style -> `m_foo(foo)`

File: Base/res/html/misc/border-radius.html [hunk 3]
```diff
@@ -239,6 +262,18 @@ <h3>Border-radius</h3>
...
+    <em>In a nutshell</em>
+    <div class="nut"></div>
+    <br>
+    <em>In a circle</em>
+    <div class="screenshot"></div>
+    <br>
+    <em>Rounded</em>
+    <div class="rounded-image"></div>
+    <br>
+    <br/>
```
Comment (linusg) [inline Base/res/html/misc/border-radius.html:275]: no `/`

File: Base/res/html/misc/border-radius.html [hunk 4]
```diff
@@ -274,6 +274,11 @@ <h3>Border-radius</h3>
     <br>
     <br/>
 
+    <p>This is an <code>&lt;img&gt;</code> tag with a border radius</p>
```
Comment (linusg) [inline Base/res/html/misc/border-radius.html:277]: since you say `an ` here, you should also do that in the commit message :)

**Review by linusg — COMMENTED** [review]
> Would you look at that? Eight commits, interesting :thinking:

What a coincidence :thinking:

File: Base/res/html/misc/border-radius.html (hunk 4 above)
Comment (MacDue) [reply Base/res/html/misc/border-radius.html:277]: Change the commit message to be consistent with the previous examples commit, so still no `an` :)

**Pull request merged**